src/Catalyst.Core.Modules.Dfs/IpfsEngine.cs

```diff
@@ -0,0 +1,673 @@
+            };
+#pragma warning disable CS1998
+            stopTasks.Add(async () => autodialer.Dispose());
+#pragma warning restore CS1998
+
+            tasks = new List<Func<Task>>
+            {
+                // Bootstrap discovery
+                async () =>
+                {
+                    var bootstrap = new Bootstrap
+                    {
+                        Addresses = await Bootstrap.ListAsync()
+                    };
+                    bootstrap.PeerDiscovered += OnPeerDiscovered;
+                    stopTasks.Add(async () => await bootstrap.StopAsync().ConfigureAwait(false));
+                    await bootstrap.StartAsync().ConfigureAwait(false);
+                },
+
+                // New multicast DNS discovery
+                async () =>
+                {
+                    if (Options.Discovery.DisableMdns)
+                        return;
+                    var mdns = new MdnsNext
+                    {
+                        LocalPeer = localPeer,
+                        MulticastService = multicast
+                    };
+                    if (Options.Swarm.PrivateNetworkKey != null)
+                        mdns.ServiceName = $"_p2p-{Options.Swarm.PrivateNetworkKey.Fingerprint().ToHexString()}._udp";
+                    mdns.PeerDiscovered += OnPeerDiscovered;
+                    stopTasks.Add(async () => await mdns.StopAsync().ConfigureAwait(false));
+                    await mdns.StartAsync().ConfigureAwait(false);
+                },
+
+                // Old style JS multicast DNS discovery
+                async () =>
+                {
+                    if (Options.Discovery.DisableMdns || Options.Swarm.PrivateNetworkKey != null)
+                        return;
+                    var mdns = new MdnsJs
+                    {
+                        LocalPeer = localPeer,
+                        MulticastService = multicast
+                    };
+                    mdns.PeerDiscovered += OnPeerDiscovered;
+                    stopTasks.Add(async () => await mdns.StopAsync().ConfigureAwait(false));
+                    await mdns.StartAsync().ConfigureAwait(false);
+                },
+
+                // Old style GO multicast DNS discovery
+                async () =>
+                {
+                    if (Options.Discovery.DisableMdns || Options.Swarm.PrivateNetworkKey != null)
+                        return;
+                    var mdns = new MdnsGo
+                    {
+                        LocalPeer = localPeer,
+                        MulticastService = multicast
+                    };
+                    mdns.PeerDiscovered += OnPeerDiscovered;
+                    stopTasks.Add(async () => await mdns.StopAsync().ConfigureAwait(false));
+                    await mdns.StartAsync().ConfigureAwait(false);
+                },
+                async () =>
+                {
+                    if (Options.Discovery.DisableRandomWalk)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/app/catalyst-network/Catalyst.Framework/pullRequest?prid=4612924)